watchtower-plugin/src/main.rs

```diff
@@ -0,0 +1,375 @@
+                                e.error,
+                                e.error_code
+                            );
+                            plugin
+                                .state()
+                                .lock()
+                                .unwrap()
+                                .add_invalid_appointment(tower_id, &appointment);
+                        }
+                    },
+                    AddAppointmentError::SignatureError(proof) => {
+                        log::warn!("Cannot recover known tower_id from the appointment receipt. Flagging tower as misbehaving");
+                        plugin
+                            .state()
+                            .lock()
+                            .unwrap()
+                            .flag_misbehaving_tower(tower_id, proof)
+                    }
+                },
+            };
+        } else if status.is_misbehaving() {
+            log::warn!(
+                "{} is misbehaving. Not sending any further appointments",
+                tower_id
+            );
+        } else {
+            if status.is_subscription_error() {
+                log::warn!(
+                    "There is a subscription issue with {}. Adding appointment to pending",
+                    tower_id,
+                );
+            } else {
+                log::warn!("{} is {}. Adding appointment to pending", tower_id, status);
+            }
+
+            plugin
+                .state()
+                .lock()
+                .unwrap()
+                .add_pending_appointment(tower_id, &appointment);
+        }
+    }
+
+    // FIXME: Ask cdecker: Do hooks need to return something?
+    Ok(json!(r#" {"result": continue}"#))
+}
+
+#[tokio::main]
+async fn main() -> Result<(), Error> {
+    let data_dir = match env::var("TOWERS_DATA_DIR") {
+        Ok(v) => PathBuf::from(v),
+        Err(_) => home_dir().unwrap().join(".watchtower"),
+    };
+
+    let builder = Builder::new(
+        Arc::new(Mutex::new(WTClient::new(data_dir).await)),
+        stdin(),
+        stdout(),
+    )
+    .option(ConfigOption::new(
+        "watchtower-port",
+        Value::Integer(9814),
+        "tower API port",
+    ))
+    .rpcmethod(
+        "registertower",
+        "Registers the client public key (user id) with the tower.",
+        register,
+    )
+    .rpcmethod(
+        "getappointment",
+        "Gets appointment data from the tower given the tower id and the locator.",
+        get_appointment,
+    )
+    .rpcmethod("listtowers", "Lists all registered towers.", list_towers)
+    .rpcmethod(
+        "gettowerinfo",
+        "Shows the info about a given tower.",
+        get_tower_info,
+    )
+    .rpcmethod(
+        "retrytower",
+        "Retries to send pending appointment to an unreachable tower.",
+        retry_tower,
+    )
+    .hook("commitment_revocation", on_commitment_revocation);
+
+    let plugin = builder.start().await.unwrap().unwrap();
```

Comment:
When running `lightningd --help` while the watchtower plugin is enabled, you get a panic while unwrapping a `None` (the second unwrap)
```
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', watchtower-plugin/src/main.rs:373:49
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
this can be handled with something like this
```rust
if let Some(plugin) = builder.start.await.unwrap() {
    plugin.join().await
} else {
    Ok(())
}
```

Reply:
Shouldn't it be


```rust
  if let Some(plugin) = builder.start().await? {
      plugin.join().await
  } else {
      Ok(())
  }
```